File: SAMSUI/src/app/Model/master/item-type.ts
import { BaseModel } from "../base/baseModel";

export class ItemTypeModel extends BaseModel{

   public orgId ?: number;

   public itemTypeDesc ?: string;

   public active ?: boolean;

   public itemTypeId ?: number;

   public itemTypeName ?: string;

   public createdBy ?:string;

   public createdDtDisp ?: string;

   public updatedBy ?: string;

   public updatedDtDisp ?: string;

   public pageNumber ?: number;

   public recordsPerPage ?: number;

   public orgName ?: string;

   public isListEmpty ?:boolean;

   public createdDt ?:Date;
   
   public updatedDt ?: Date;

   public systemGenerated ?:boolean;
}

File: sams-backend/src/main/java/com/sams/entity/ItemType.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_item_type", schema = "inventory")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ItemType {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "org_id")
    private Long orgId;

    @Column(name = "item_type_desc")
    private String itemTypeDesc;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "item_type_id")
    private Long itemTypeId;

    @Column(name = "item_type_name")
    private String itemTypeName;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @Column(name = "org_name")
    private String orgName;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

    @Column(name = "system_generated")
    private Boolean systemGenerated;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}